From e6ede70d8218ebc031f94bb93e4144b198bb7789 Mon Sep 17 00:00:00 2001
From: Judith Mendez <jm@ti.com>
Date: Mon, 2 Feb 2026 13:59:26 -0600
Subject: [PATCH 4/7] PENDING: mmc: sdhci_am654: Allow defaults for DLL
 properties

DT properties ti,trm-icp and ti,driver-strength-ohm seem to use
consistent values across TI SoC's. So instead of failing probe if
these values are missing, set a sane default.

Signed-off-by: Judith Mendez <jm@ti.com>
Reviewed-by: Moteen Shah <m-shah@ti.com>
---
 drivers/mmc/host/sdhci_am654.c | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/host/sdhci_am654.c b/drivers/mmc/host/sdhci_am654.c
index 21804148426f..db94df574846 100644
--- a/drivers/mmc/host/sdhci_am654.c
+++ b/drivers/mmc/host/sdhci_am654.c
@@ -78,6 +78,10 @@
 #define ITAPCHGWIN_SHIFT	9
 #define ITAPCHGWIN_MASK		BIT(ITAPCHGWIN_SHIFT)
 
+#define DEFAULT_DLL_TRIM_ICP	8
+
+#define DEFAULT_DLL_DRIVER_STRENGTH	50
+
 #define DRIVER_STRENGTH_50_OHM	0x0
 #define DRIVER_STRENGTH_33_OHM	0x1
 #define DRIVER_STRENGTH_66_OHM	0x2
@@ -871,13 +875,19 @@ static int sdhci_am654_get_of_property(struct platform_device *pdev,
 	if (sdhci_am654->flags & DLL_PRESENT) {
 		ret = device_property_read_u32(dev, "ti,trm-icp",
 					       &sdhci_am654->trm_icp);
-		if (ret)
-			return ret;
+		if (ret) {
+			sdhci_am654->trm_icp = DEFAULT_DLL_TRIM_ICP;
+			dev_warn(dev, "ti,trm-icp not found, using default value: 0x%x\n",
+				 sdhci_am654->trm_icp);
+		}
 
 		ret = device_property_read_u32(dev, "ti,driver-strength-ohm",
 					       &drv_strength);
-		if (ret)
-			return ret;
+		if (ret) {
+			drv_strength = DEFAULT_DLL_DRIVER_STRENGTH;
+			dev_warn(dev, "ti,driver-strength-ohm not found, using default value: %d ohm\n",
+				 drv_strength);
+		}
 
 		switch (drv_strength) {
 		case 50:
-- 
2.53.0

